mmpose/core/evaluation/pose3d_eval.py

```diff
@@ -50,3 +50,116 @@ def keypoint_mpjpe(pred, gt, mask, alignment='none'):
     error = np.linalg.norm(pred - gt, ord=2, axis=-1)[mask].mean()
 
     return error
+
+
+def keypoint_3d_pck(pred, gt, mask, alignment='none', threshold=0.15):
+    """Calculate the Percentage of Correct Keypoints (3DPCK) w. or w/o rigid
+    alignment.
+
+    Paper ref: `Monocular 3D Human Pose Estimation In The Wild Using Improved
+    CNN Supervision' 3DV`2017
+    More details can be found in the `paper
+    <https://arxiv.org/pdf/1611.09813>`__.
+
+    batch_size: N
+    num_keypoints: K
+    keypoint_dims: C
+
+    Args:
+        pred (np.ndarray[N, K, C]): Predicted keypoint location.
+        gt (np.ndarray[N, K, C]): Groundtruth keypoint location.
+        mask (np.ndarray[N, K]): Visibility of the target. False for invisible
+            joints, and True for visible. Invisible joints will be ignored for
+            accuracy calculation.
+        alignment (str, optional): method to align the prediction with the
+            groundtruth. Supported options are:
+            - ``'none'``: no alignment will be applied
+            - ``'scale'``: align in the least-square sense in scale
+            - ``'procrustes'``: align in the least-square sense in scale,
+                rotation and translation.
+        threshold:  If L2 distance between the prediction and the groundtruth
+            is less then threshold, the predicted result is considered as
+            correct. Default: 0.15 (m).
+    Returns:
+        pck: percentage of correct keypoints.
+    """
+    assert mask.any()
```

Comment:
The "pred" and "gt" are single-person or multi-person keypoints? Is it a possible case that the mask is all False?

Reply:
They are single-person keypoints. The mask here can't be all False. For multi-person pose estimation, I think maybe we can add a post-processing step to evaluate every single person respectively.